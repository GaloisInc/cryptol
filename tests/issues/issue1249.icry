// toSignedInteger

toSignedInteger (0 : [64])
:eval toSignedInteger (0 : [64])
toSignedInteger (1 : [64])
:eval toSignedInteger (1 : [64])
toSignedInteger (2^^63 - 1 : [64])
:eval toSignedInteger (2^^63 - 1 : [64])
toSignedInteger (-1 : [64])
:eval toSignedInteger (-1 : [64])
toSignedInteger (-(2^^63) : [64])
:eval toSignedInteger (-(2^^63) : [64])

// deepseq

deepseq ((), undefined`{()}) True
:eval deepseq ((), undefined`{()}) True
