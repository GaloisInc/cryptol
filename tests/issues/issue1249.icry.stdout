Loading module Cryptol
0
0
1
1
9223372036854775807
9223372036854775807
-1
-1
-9223372036854775808
-9223372036854775808

Run-time error: undefined
-- Backtrace --
Cryptol::error called at Cryptol:1062:13--1062:18
Cryptol::undefined called at issue1249.icry:16:14--16:23
Cryptol::deepseq called at issue1249.icry:16:1--16:8
Run-time error: undefined
